Project/Controllers/ToDosController.cs

```diff
@@ -87,7 +84,94 @@ public ActionResult<IEnumerable<ToDos>> GetToDoItems()
             }
             catch (Exception e)
             {
-                return StatusCode(StatusCodes.Status500InternalServerError,
+                return StatusCode(StatusCodes.Status404NotFound,
+                    new
+                    {
+                        Message = e.InnerException != null
+                            ? $"{e.Message} {e.InnerException.Message}"
+                            : e.Message
+                    });
+            }
+        }
+
+        [HttpPut("{id}")]
+        [ProducesDefaultResponseType]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult PutToDoItem(int id, ToDos todos)
+        {
+            try
+            {
+                if (id != todos.Id)
+                {
+                    return BadRequest();
+                }
+
+                _context.Entry(todos).State = EntityState.Modified;
+                _context.SaveChanges();
+
+                return CreatedAtAction("GetToDoItem", new ToDos {Id = todos.Id}, todos);
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status404NotFound,
+                    new
+                    {
+                        Message = e.InnerException != null
+                            ? $"{e.Message} {e.InnerException.Message}"
+                            : e.Message
+                    });
+            }
+        }
+
+        [HttpDelete("{id}")]
+        [ProducesDefaultResponseType]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public ActionResult<ToDos> DeleteToDoItem(int id)
+        {
+            try
+            {
+                var toDosItem = _context.ToDosItems.Find(id);
+
+                if (toDosItem == null)
+                {
+                    return NotFound();
+                }
+
+                _context.ToDosItems.Remove(toDosItem);
+                _context.SaveChanges();
+
+                return toDosItem;
+            }
+            catch (Exception e)
+            {
+                return StatusCode(StatusCodes.Status404NotFound,
+                    new
+                    {
+                        Message = e.InnerException != null
+                            ? $"{e.Message} {e.InnerException.Message}"
+                            : e.Message
+                    });
```

Comment:
Apart from my previous comment about generic exceptions, this code is very repetitive, you should consider refactoring it.

Reply:
You never want an API to return 500 -- meaning it broke. Need to return specific codes / reasons so that clients can try to remedy the error (validation for example / auth )